tests/UnitTests/TestContainer.cpp

```diff
@@ -0,0 +1,163 @@
+//===- TestContainer.h ----------------------------------------------------===//
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements the tests for the MemRef class.
+//
+//===----------------------------------------------------------------------===//
+
+#include "Test.h"
+#include "Utils/Container.h"
+
+// Fixture for testing the MemRef class.
+class MemRefTest : public ::testing::Test {
+protected:
+  MemRefTest()
+      : m1(MemRef<float, 1>({2})), m2(MemRef<float, 2>({2, 3})),
+        m3(MemRef<float, 3>({2, 3, 4})), m4(MemRef<float, 4>({2, 3, 4, 5})) {}
+
+  void SetUp() override {
+    std::iota(m3.getData(), m3.getData() + 2, 0.);
+    std::iota(m2.getData(), m2.getData() + 6, 0.);
+    std::iota(m3.getData(), m3.getData() + 24, 0.);
+    std::iota(m4.getData(), m4.getData() + 120, 0.);
+  }
+
+  MemRef<float, 1> m1;
+  MemRef<float, 2> m2;
+  MemRef<float, 3> m3;
+  MemRef<float, 4> m4;
+};
+
+// 1 dimensional memref.
+TEST_F(MemRefTest, 1DMemref) {
+  intptr_t true_strides[] = {1};
+  ASSERT_ARRAY_EQ(m1.getStrides(), true_strides);
+}
+
+// 2 dimensional memref.
+TEST_F(MemRefTest, 2DMemref) {
+  intptr_t true_strides[] = {3, 1};
```

Comment:
For the issue above, I suggest doing the following things, if you have a more elegant way, feel free to follow your idea.
- Use `sizeof(true_strides)/sizeof(intptr_t)` to get the length of the `true_strides`.
- Define a function to class MemRef named `getStridesLength` (or something like this) to return the length.
- Compare this two length.
- Pass the length to `ASSERT_ARRAY_EQ`.

Reply:
Done, thanks a lot.

I've added two member functions to the MemRef class:
- `getSize` returns the number of elements.
- `getRank` returns the number of dimensions.